src/XperienceCommunity.Localization.Base/Localizer/XperienceStringLocalizerBase.cs

```diff
@@ -0,0 +1,182 @@
+﻿using AngleSharp.Common;
+using CMS.ContentEngine;
+using CMS.DataEngine;
+using CMS.Helpers;
+using CMS.Websites.Routing;
+using Microsoft.Extensions.Localization;
+using System.Data;
+using XperienceCommunity.Localization;
+
+namespace XperienceCommunity.Localizer.Internal
+{
+    public class XperienceStringLocalizerBase(IProgressiveCache progressiveCache,
+        IWebsiteChannelContext websiteChannelContext,
+        IInfoProvider<ContentLanguageInfo> contentLanguageInfoProvider)
+    {
+        private readonly IProgressiveCache progressiveCache = progressiveCache;
+        private readonly IWebsiteChannelContext websiteChannelContext = websiteChannelContext;
+        private readonly IInfoProvider<ContentLanguageInfo> contentLanguageInfoProvider = contentLanguageInfoProvider;
```

Comment:
With primary constructor these lines are not needed. Can you remove them?

Reply:
I'll do this, but it can be argued that still declaring readonly private members is a better way than just leaving them in the constructor.  I've had this talk with Sean before, but i'll do it for ya.

Reply:
(I usually also use the _ prefix for saying this is an internal private field)